### PR TITLE
Add arguments to RechartsFunction

### LINKED ISSUE #20722: [@types/recharts] RechartsFunction has no arguments
- [x] I tried using the `@types/recharts` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @mthmulders @rapmue @royxue 

The RechartsFunction which is used in all components onClick callbacks has no arguments, whereas it should have data, index and event arguments.

I guess it should be something like this:
`RechartsFunction = (data: any, index: number, event: MouseEvent) => void`

### PR DESCRIPTION
**Pull request by mthmulders**
Resolves #20722.

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20722#issuecomment-355878319](original GH issue)
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/recharts/index.d.ts [hunk 1]
```diff
@@ -375,7 +371,7 @@ export interface LegendProps {
     onClick?: RechartsFunction;
     onMouseEnter?: RechartsFunction;
     onMouseLeave?: RechartsFunction;
-    onBBoxUpdate?: (box: { width: number; height: number; }) => void;
+    onBBoxUpdate?: RechartsFunction;
```
Comment (rapmue) [inline types/recharts/index.d.ts]: hm, I'm not sure about this one. Is this one also overloaded? In the source code its either a box or null -> https://github.com/recharts/recharts/blob/9d42367e658cab6bed0a8f2517f17be9e0cf0957/src/component/Legend.js on line 150 resp. 160

**Review by rapmue — COMMENTED** [review]
I think everything looks good except what I commented about onBBoxUpdate

File: types/recharts/index.d.ts (hunk 1 above)
Comment (mthmulders) [reply types/recharts/index.d.ts]: Good catch! My idea is that, whatever method argument we have, as long as the return type is `void` we could use `RechartsFunction`. In its current form, `dtslint` says an inline type definition is not the way the go, so I needed to change this line anyway.

**Comment by weswigham** [comment]
@mthmulders do you plan on making any more changes or is this good to go?

**Comment by rapmue** [comment]
@mthmulders a sorry, I missed that you are waiting for my reply. Personally I would make a seperate type for this one, since it's clear what the accepted values are.
Thanks for your work!

**Comment by weswigham** [comment]
Good now?

**Comment by mthmulders** [comment]
I think the final say is on @rapmue.

**Comment by andy-ms** [comment]
Merging but these should eventually not all share the same type.

**Pull request merged**